### PR TITLE
add setting for ascii grid symbols

### PR DESCRIPTION
**Pull request by s-mx**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
This setting allows to chose charset for printing grids (either utf8 or
ascii).

Detailed description / Documentation draft:

Allows to change a charset which is used for printing grids borders. Available charsets are following: UTF-8, ASCII.

**Example** 

```text
SET output_format_pretty_grid_charset = 'UTF-8';
SELECT * FROM a;
┌─a─┐
│ 1 │
└───┘
SET output_format_pretty_grid_charset = 'ASCII';
SELECT * FROM a;
+-a-+
| 1 |
+---+
```

### CODE REVIEW

**Comment by s-mx** [comment]
#10429

File: src/Core/Settings.h [hunk 1]
```diff
@@ -438,6 +438,7 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingUInt64, output_format_pretty_max_column_pad_width, 250, "Maximum width to pad all values in a column in Pretty formats.", 0) \
     M(SettingUInt64, output_format_pretty_max_value_width, 10000, "Maximum width of value to display in Pretty formats. If greater - it will be cut.", 0) \
     M(SettingBool, output_format_pretty_color, true, "Use ANSI escape sequences to paint colors in Pretty formats", 0) \
+    M(SettingString, output_format_pretty_grid_charset, "utf8", "Charset for printing grid borders. Available charsets: ascii, utf8 (default one).", 0) \
```
Comment (alexey-milovidov) [inline src/Core/Settings.h:441]: Ok, but UTF-8 is written as `UTF-8` (with dash).

Why it is important?
Some tools (e.g. Perl) have both variants - UTF8 and UTF-8 that have different meanings.

See https://en.wikipedia.org/wiki/UTF-8 (official name and variants).

Also let's write them in capitals: ASCII, UTF-8.

File: src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp [hunk 2]
```diff
@@ -106,38 +154,42 @@ void PrettyBlockOutputFormat::write(const Chunk & chunk, PortKind port_kind)
     Widths name_widths;
     calculateWidths(header, chunk, widths, max_widths, name_widths);
 
+    const GridSymbols & gridSymbols = format_settings.pretty.charset == FormatSettings::Pretty::Charset::UTF8 ?
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:157]: `grid_symbols` for style.

File: tests/queries/0_stateless/00405_pretty_formats.reference [hunk 3]
```diff
@@ -211,3 +211,96 @@ Showed first 6.
      5   5       (5,'5')                 2 
 
 Showed first 6.
++-------+-------+---------+-----------------+
+| [1mhello[0m | [1mworld[0m | [1mtuple  [0m | [1msometimes_nulls[0m |
++-------+-------+---------+-----------------+
+|     0 | 0     | (0,'0') |            ᴺᵁᴸᴸ |
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/00405_pretty_formats.reference:217]: It's funny that we use special unicode symbols for ᴺᵁᴸᴸ even for ASCII variant.
Maybe also use ASCII if user requested that?

Comment (alexey-milovidov) [reply tests/queries/0_stateless/00405_pretty_formats.reference:217]: And also there are some symbols like &hellip;

**Comment by s-mx** [comment]
Hi @qoega! Could you clarify what did I miss in documentation of new setting?

**Comment by qoega** [comment]
It is just a label for further work with docs. Translate it, etc.

File: src/Core/Settings.h (hunk 1 above)
Comment (s-mx) [reply src/Core/Settings.h]: Made them in capitals.

File: tests/queries/0_stateless/00405_pretty_formats.reference (hunk 3 above)
Comment (s-mx) [reply tests/queries/0_stateless/00405_pretty_formats.reference]: Added replacements for ᴺᵁᴸᴸ and …

File: src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp [hunk 4]
```diff
@@ -222,10 +282,16 @@ void PrettyBlockOutputFormat::writeValueWithPadding(
         serialized_value.resize(UTF8::computeBytesBeforeWidth(
             reinterpret_cast<const UInt8 *>(serialized_value.data()), serialized_value.size(), 0, 1 + format_settings.pretty.max_value_width));
 
+        const char *ellipsis = format_settings.pretty.charset == FormatSettings::Pretty::Charset::UTF8 ?
+                               "⋯" : "...";
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:286]: But it takes three columns instead of one in terminal...

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
> Yandex synchronization check (only for Yandex employees) 

It's a bug in Arcadia Devtools @exprmntr

**Pull request merged**